consumer/consumer_test.go

```diff
@@ -169,141 +169,432 @@ func TestParseMessageWithImproperReport(t *testing.T) {
+	assert.False(t, shouldProcess, "checkReportStructure should return shouldProcess = false for empty reports")
+}
+
+func TestCheckReportStructureReportWithAllAttributesPresentAndEmpty(t *testing.T) {
+	deserialized, err := consumer.DeserializeMessage([]byte(testdata.ConsumerMessage))
+	assert.Nil(t, err, "deserializeMessage should not return error for empty report")
+
+	shouldProcess, err := consumer.CheckReportStructure(*deserialized.Report)
+	assert.Nil(t, err, "checkReportStructure should return err = nil for empty reports")
+	assert.False(t, shouldProcess, "checkReportStructure should return"+
+		" shouldProcess = false for empty reports where all expected attributes are present")
+}
+
+// If some attributes are missing, but all the present attributes are empty, we just
+// skip the processing of the message.
+func TestCheckReportStructureReportWithEmptyAndMissingAttributes(t *testing.T) {
+	message := `{
+		"OrgID": ` + fmt.Sprint(testdata.OrgID) + `,
+		"ClusterName": "` + string(testdata.ClusterName) + `",
+		"Report": {
+			"fingerprints": [],
+			"reports": [],
+			"skips": []
+		}
+	}`
+
+	deserialized, err := consumer.DeserializeMessage([]byte(message))
+	assert.Nil(t, err, "deserializeMessage should not return error for empty report")
+
+	shouldProcess, err := consumer.CheckReportStructure(*deserialized.Report)
+	assert.Nil(t, err, "checkReportStructure should return err = nil for empty reports")
+	assert.False(t, shouldProcess, "checkReportStructure should return shouldProcess = false for empty reports")
+}
+func TestCheckReportStructureReportWithItems(t *testing.T) {
+	message := `{
+		"OrgID": ` + fmt.Sprint(testdata.OrgID) + `,
+		"ClusterName": "` + string(testdata.ClusterName) + `",
+		"Report": ` + string(testdata.Report2Rules) + `
+	}`
+
+	deserialized, err := consumer.DeserializeMessage([]byte(message))
+	assert.Nil(t, err, "deserializeMessage should not return error for empty report")
+
+	shouldProcess, err := consumer.CheckReportStructure(*deserialized.Report)
+	assert.Nil(t, err, "checkReportStructure should return err = nil for empty reports")
+	assert.True(t, shouldProcess, "checkReportStructure should return shouldProcess = true for reports with items")
+}
+
+func TestParseReportContentEmptyReportsAttribute(t *testing.T) {
+	deserialized, err := consumer.DeserializeMessage([]byte(testdata.ConsumerMessage))
+	assert.Nil(t, err, "deserializeMessage should not return error for this message")
+
+	err = consumer.ParseReportContent(&deserialized)
+	assert.Nil(t, err, "ParseReportContent should not return nil for empty report")
+}
+
+func TestParseReportContentValidReport(t *testing.T) {
 	message := `{
+		"OrgID": ` + fmt.Sprint(testdata.OrgID) + `,
+		"ClusterName": "` + string(testdata.ClusterName) + `",
+		"Report":` + string(testdata.Report2Rules) + `
+	}`
+
+	//TODO: replace call to deserialize with exported function that returns incomingMessage
```

Comment:
have you missed this TODO?

Reply:
Yep ^^

Reply:
will remove the todo. The trick works, but it's really messy code